pkg/asset/machines/baremetal/hosts.go

```diff
@@ -66,6 +66,9 @@ func Hosts(config *types.InstallConfig, machines []machineapi.Machine) (*HostSet
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      host.Name,
 				Namespace: "openshift-machine-api",
+				// Pause reconciliation until we can annotate with the initial
+				// status containing the HardwareDetails
+				Annotations: map[string]string{"baremetalhost.metal3.io/paused": ""},
```

Comment:
This annotation should only be set on masters. Moving it down to the block starting on line 84 should accomplish that.